src/bundle/Resources/config/services.yml

```diff
@@ -1,9 +1,16 @@
 parameters:
     ezplatform.cron.registry.cronjobs.class: EzSystems\EzPlatformCronBundle\Registry\CronJobsRegistry
+    ezplatform.console.command.run.class: EzSystems\EzPlatformCronBundle\Command\CronRunCommand
 
 services:
     ezplatform.cron.registry.cronjobs:
         class: '%ezplatform.cron.registry.cronjobs.class%'
         arguments:
             - '%kernel.environment%'
             - '@ezpublish.siteaccess'
+            -
+    ezplatform.console.command.run:
+        class: '%ezplatform.console.command.run.class%'
+        tags:
+            - { name: console.command }
+            - { name: monolog.logger, channel: cronjob }
```

Comment:
You don't need it anymore.
 - `LoggerInterface $cronjobLogger`,
 - `LoggerAwareInterface` + this tag

are two ways of injecting the right channel. I wasn't aware of previous answer and it is "either/or" rather than do both.

Reply:
@Nattfarinn Autowiring (https://symfony.com/doc/current/logging/channels_handlers.html#how-to-autowire-logger-channels) doesn't work with Symfony 3.4. Consequently the first variant (`LoggerInterface $cronjobLogger`) is out I think.

For the second variant I need some help. Are the following steps correct:
- Class CronRunCommand must implement LoggerAwareInterface
- setLogger function has to be implemented for class CronRunCommand
- the service is configured with monolog.logger tag
- the service calls setLogger with specific logger channel

Reply:
@homarx Exactly, and the last step is done automatically during container build by Symfony.

Reply:
@Nattfarinn It doesn't work without the setLogger call configured in service. 

CronRunCommand.php:
```
<?php

/**
 * @copyright Copyright (C) Ibexa AS. All rights reserved.
 * @license For full copyright and license information view LICENSE file distributed with this source code.
 */
namespace EzSystems\EzPlatformCronBundle\Command;

use Cron\Cron;
use Cron\Executor\Executor;
use Cron\Resolver\ArrayResolver;
use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
use Symfony\Component\Console\Input\InputInterface;
use Symfony\Component\Console\Input\InputOption;
use Symfony\Component\Console\Output\OutputInterface;
use Psr\Log\LoggerAwareInterface;
use Psr\Log\LoggerInterface;

class CronRunCommand extends ContainerAwareCommand implements LoggerAwareInterface
{
    /**
     * @var \Psr\Log\LoggerInterface
     */
    private $logger;

    /**
     * @param \Psr\Log\LoggerInterface $logger
     */
    public function setLogger(LoggerInterface $logger)
    {
        $this->logger = $logger;
    }

    protected function configure()
    {
        $this
            ->setDefinition([
                new InputOption('category', null, InputOption::VALUE_REQUIRED, 'Job category to run', 'default'),
            ])
            ->setName('ezplatform:cron:run')
            ->setDescription('Perform one-time cron tasks run.')
            ->setHelp(
                <<<EOT
It's not meant to be run manually, yet it's OK to do so as it still might be useful for development purpose.

Check documentation how to setup it be called automatically.
EOT
            );
    }

    protected function execute(InputInterface $input, OutputInterface $output)
    {
        $registry = $this->getContainer()->get('ezplatform.cron.registry.cronjobs');

        $category = $input->getOption('category');
        $cronJobs = $registry->getCategoryCronJobs($category);

        $resolver = new ArrayResolver($cronJobs);

        $cron = new Cron();
        $cron->setExecutor(new Executor());
        $cron->setResolver($resolver);

        $reports = $cron->run();

        while ($cron->isRunning()) {
            usleep(10000);
        }

        if ($this->logger) {
            foreach ($reports->getReports() as $report) {
                $process = $report->getJob()->getProcess();
                $extraInfo = [
                    'command' => $process->getCommandLine(),
                    'exitCode' => $process->getExitCode(),
                ];
                foreach ($report->getOutput() as $reportOutput) {
                    $this->logger->info($reportOutput, $extraInfo);
                }
                if (!$report->isSuccessful()) {
                    foreach ($report->getError() as $reportError) {
                        $reportError = trim($reportError);
                        if (!empty($reportError)) {
                            $this->logger->error($reportError, $extraInfo);
                        }
                    }
                }
            }
        }
    }
}
```

service.yml:
```
parameters:
    ezplatform.cron.registry.cronjobs.class: EzSystems\EzPlatformCronBundle\Registry\CronJobsRegistry
    ezplatform.console.command.run.class: EzSystems\EzPlatformCronBundle\Command\CronRunCommand

services:
    ezplatform.cron.registry.cronjobs:
        class: '%ezplatform.cron.registry.cronjobs.class%'
        arguments:
            - '%kernel.environment%'
            - '@ezpublish.siteaccess'

    ezplatform.console.command.run:
        class: '%ezplatform.console.command.run.class%'
#        calls:
#            - [setLogger, ["@monolog.logger.cronjob"]]
        tags:
            - { name: console.command }
            - { name: monolog.logger, channel: cronjob }
```

When I comment in the setLogger call in service.yml, logfile is written as expected.

I'm implementing on Platform v2 with Symfony 3.4

Reply:
Huh, my bad. I was almost 100% sure it was already part of 3.4, but turns out it is not.

Ignore the `LoggerAwareInterface` then and just go with plain old dependency injection (use `LoggerInterface $logger` in constructor):

```yaml

    ezplatform.console.command.run:
         class: '%ezplatform.console.command.run.class%'
         arguments:
             $logger: "@monolog.logger.cronjob"

Reply:
@Nattfarinn If monolog.logger.cronjob service is injected as argument it has to be configured. Otherwise an error would be thrown.
If the service is injected via tag it doesen't matter if cronjob channel is configured or not:

```
    ezplatform.console.command.run:
        class: '%ezplatform.console.command.run.class%'
        tags:
            - { name: console.command }
            - { name: monolog.logger, channel: cronjob }
```

Which do you think is the better way?

Reply:
@Nattfarinn I choosed the injection via argument and added the missing config (monolog channel). Can you please take a look at my last commit?